test/Test/Main.purs

```diff
@@ -0,0 +1,7 @@
+module Test.Main where
+
+import Test.S
+
+main = do
```

Comment:
It would be nice to get a StrongCheck test which can verify that any arbitrary search ADT can be pretty-printed to a search query which can be parsed to the same search ADT.